core/codenvy-hosted-api-admin/src/main/java/com/codenvy/api/dao/ldap/AdminUserDaoImpl.java

```diff
@@ -64,19 +74,62 @@ public AdminUserDaoImpl(AccountDao accountDao,
               userAttributesMapper,
               eventService);
         this.userLdapPagination = userLdapPagination;
+        final StringBuilder sb = new StringBuilder();
+        for (String objectClass : userAttributesMapper.userObjectClasses) {
+            sb.append("(objectClass=");
+            sb.append(objectClass);
+            sb.append(')');
+        }
+        this.objectClassFilter = sb.toString();
     }
 
-
     @Override
-    public List<User> getAll(int maxItems, int skipCount) throws ServerException, IllegalArgumentException {
+    public Page<User> getAll(int maxItems, int skipCount) throws ServerException, IllegalArgumentException {
         checkArgument(maxItems >= 0, "The number of items to return can't be negative.");
         checkArgument(skipCount >= 0, "The number of items to skip can't be negative.");
 
         try {
-            return userLdapPagination.get(maxItems, skipCount);
+            return new Page<>(userLdapPagination.get(maxItems, skipCount), skipCount, maxItems, getTotalUsersCount());
         } catch (NamingException | IOException e) {
             LOG.error(e.getMessage(), e);
             throw new ServerException("Unable get all users.", e);
         }
     }
+
+    // TODO: consider WeakReference holder for returned value.
+    // as this operation may be expensive on large medium/large data sets
+    private int getTotalUsersCount() throws ServerException {
+        InitialLdapContext context = null;
+        try {
+            context = contextFactory.createContext();
+            final SearchControls controls = new SearchControls();
+            controls.setSearchScope(SearchControls.SUBTREE_SCOPE);
+            controls.setReturningAttributes(EMPTY_ARRAY);
```

Comment:
yes, there is no way to count elements using ldap filters, the only way is to count elements in enumeration, so it is much cheaper to return N matched elements which are empty and count them.